hs_geographic_feature_resource/models.py

```diff
@@ -0,0 +1,390 @@
+from django.db import models
+from mezzanine.pages.page_processors import processor_for
+from hs_core.models import BaseResource, ResourceManager, resource_processor, CoreMetaData, AbstractMetaDataElement
+from mezzanine.pages.models import Page
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from django.contrib.contenttypes import generic
+
+# Create your models here.
+
+
+class OriginalFileInfo(AbstractMetaDataElement):
+
+    term = 'OriginalFileInfo'
+
+    fileTypeEnum=(  (None, 'Unknown'),
+                    ("SHP", "ESRI Shapefiles"),
+                    ("ZSHP", "Zipped ESRI Shapefiles"),
+                    ("KML", "KML"),
+                    ("KMZ", "KMZ"),
+                    ("GML", "GML"),
+                    ("SQLITE", "SQLite")
+                 )
+    fileType=models.TextField(max_length=128, choices=fileTypeEnum, default=None)
+    baseFilename=models.TextField(max_length=256, null=False, blank=False)
+    fileCount=models.IntegerField(null=False, blank=False, default=0)
+
+    filenameString = models.TextField(max_length=2048, null=True, blank=True)
+
+    class Meta:
+        # OriginalFileInfo element is not repeatable
+        unique_together = ("content_type", "object_id")
+
+    @classmethod
+    def create(cls, **kwargs):
+
+        for limit in ['fileType', 'baseFilename', 'fileCount']:
+            if not limit in kwargs:
+                raise ValidationError("For OriginalFileInfo meta, one or more attribute is missing.")
+
+        metadata_obj = kwargs['content_object']
+        ori_file_info = OriginalFileInfo.objects.create(fileType=kwargs['fileType'], baseFilename=kwargs['baseFilename'],
+                                                  fileCount=kwargs['fileCount'],
+                                                 content_object=metadata_obj,)
+        # save filenameString info
+        for key, value in kwargs.iteritems():
+            if key in ('filenameString'):
+                setattr(ori_file_info, key, value)
+                ori_file_info.save()
+
+        return ori_file_info
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        ori_file_info = OriginalFileInfo.objects.get(id=element_id)
+        if ori_file_info:
+            # save filenameString info
+            for key, value in kwargs.iteritems():
+                if key in ('fileType', 'baseFilename', 'fileCount', 'filenameString'):
+                    setattr(ori_file_info, key, value)
+                    ori_file_info.save()
+        else:
+            raise ObjectDoesNotExist("No OriginalFileInfo was found for the provided id: %d" % element_id)
+
+    @classmethod
+    def remove(cls, element_id):
+        ori_file_info = OriginalFileInfo.objects.get(id=element_id)
+        if ori_file_info:
+            ori_file_info.delete()
+        else:
+            raise ObjectDoesNotExist("No OriginalFileInfo element exists for id: %d."%element_id)
```

Comment:
More explanation: There is nothing in this method that is specific to this model. It fetches the object via the model, but you can use the `cls` parameter instead. You can use `cls.__name__` in the exception message if you want to raise an exception with a model-specific message, too.

If this `remove()` method is being copied from other resources, then they are _all_ probably _broken_ too. This is totally wrong, unfortunately:

```
ori_file_info = OriginalFileInfo.objects.get(id=element_id)
if ori_file_info:
    ...
```

Because that will never be false. It looks like its expecting the Element ID might not match, but if that's the case then `get()` raises an exception, rather than returning `None` as this code seems to expect. I am afraid it looks likely that *all existing Metadata Element models are broken*.